messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/jms/JMSTopicRemove.java

```diff
@@ -53,17 +51,16 @@ private JMSTopicRemove() {
 
     protected void performRuntime(OperationContext context, ModelNode operation, ModelNode model) throws OperationFailedException {
         final ServiceName serviceName = MessagingServices.getActiveMQServiceName(PathAddress.pathAddress(operation.get(ModelDescriptionConstants.OP_ADDR)));
+        final ServiceName jmsServiceName = JMSServices.getJmsManagerBaseServiceName(serviceName);
         final String name = context.getCurrentAddress().getLastElement().getValue();
 
-        ServiceController<?> service = context.getServiceRegistry(false).getService(serviceName);
-        ActiveMQServer server = ActiveMQServer.class.cast(service.getValue());
-        JMSServerControl control = JMSServerControl.class.cast(server.getManagementService().getResource(ResourceNames.JMS_SERVER));
-        if (control != null) {
-            try {
-                control.destroyTopic(name, true);
-            } catch (Exception e) {
-                throw new OperationFailedException(e);
-            }
+        ServiceController<?> service = context.getServiceRegistry(false).getService(jmsServiceName);
```

Comment:
getServiceRegistry(true)

Reply:
fixed (by amending the WFLY-9407 commit).